source/patterns/@aws-solutions-constructs/core/lib/s3-bucket-helper.ts

```diff
@@ -56,21 +61,22 @@ export function applySecureBucketPolicy(s3Bucket: s3.Bucket): void {
       principals: [new AnyPrincipal()],
       effect: Effect.DENY,
       conditions:
-            {
-              Bool: {
-                'aws:SecureTransport': 'false'
-              }
-            }
+      {
+        Bool: {
+          'aws:SecureTransport': 'false'
+        }
+      }
     })
   );
 }
 
-export function createLoggingBucket(scope: Construct, bucketId: string, removalPolicy?: RemovalPolicy): s3.Bucket {
-  let loggingBucketProps;
+export function createLoggingBucket(scope: Construct,
```

Comment:
We've got logic for combining incoming props, defaultprops and props from the real bucket props scattered around too much. Let's make userLoggingBucketProps required (and just change the name to loggingBucketProps). Then we'll do any creation of this object upstream (since the RemovalPolicy aspect already has to be upstream).